packages/sdks-tests/src/e2e-tests/countdown.spec.ts

```diff
@@ -10,15 +10,47 @@ test.describe('Symbol with JS Code', () => {
 
     await page.goto(`/symbol-with-jscode`);
 
-    const currentSecondsOnPage = await page.locator('#seconds').textContent();
+    const getTime = async () => {
+      const secondsOnPage = await page.locator('#seconds').textContent();
+      const minutesOnPage = await page.locator('#minutes').textContent();
+      const hoursOnPage = await page.locator('#hours').textContent();
+      const daysOnPage = await page.locator('#days').textContent();
 
-    await page.waitForTimeout(2000);
-    const newSecondsOnPage = await page.locator('#seconds').textContent();
-    expect(newSecondsOnPage).not.toBe(currentSecondsOnPage);
+      return `${daysOnPage}:${hoursOnPage}:${minutesOnPage}:${secondsOnPage}`;
+    };
 
-    await page.waitForTimeout(2000);
-    const newSecondsOnPage2 = await page.locator('#seconds').textContent();
-    expect(newSecondsOnPage2).not.toBe(newSecondsOnPage);
-    expect(newSecondsOnPage2).not.toBe(currentSecondsOnPage);
+    // Wait for the countdown to start. Initial time value is 0:0:0:0
+    await expect.poll(async () => await getTime()).not.toBe('0:0:0:0');
+
+    const firstCountdownValue = await getTime();
+    let secondCountdownValue: string | undefined = undefined;
+
+    // Wait for the countdown to show a new value less than the previous value
+    await expect
+      .poll(
+        async () => {
+          secondCountdownValue = await getTime();
+          return secondCountdownValue < firstCountdownValue;
+        },
+        {
+          message: 'Make sure the countdown eventually updates correctly',
+        }
+      )
+      .toBe(true);
+
+    // repeat the process to ensure the countdown is updating correctly
+    await expect
+      .poll(
```

Comment:
discovered this polling function which will keep checking until the condition is `true`. Should be a lot less flaky than a hardcoded `page.waitForTimeout()`